rpc/errors.go

```diff
@@ -95,3 +100,15 @@ func (e *shutdownError) Code() int {
 func (e *shutdownError) Error() string {
 	return "server is shutting down"
 }
+
+// connectionClosedError is thrown when the connection is closed
+type connectionClosedError struct {
+}
+
+func (e *connectionClosedError) Code() int {
```

Comment:
> It would be something like rpc.RPCError.ErrorCode.

That's a weird way of thinking about it. The only place where the method name is
visible is when defining a method.

Compare

```go
func (someType) Code() int { ... }
```

with 

```go
func (someType) ErrorCode() int { ... }
```.

Which one tells you more about the purpose of this method?